File: py/problem_4.py
#!/usr/bin/env python


def products_of_three_digit_numbers():
    return (x * y for x in range(100, 1000) for y in range(100, 1000))


def is_palindrome(value):
    value_str = str(value)
    return value_str == value_str[::-1]


print(max(filter(is_palindrome, products_of_three_digit_numbers())))

File: dart/problem_4.dart
#!/usr/bin/env dart

import 'iterables.dart';
import 'strings.dart';

Iterable<int> productsOfThreeDigitNumbers() {
  return mapProduct(
      range(100, 1000), range(100, 1000), (int i, int j) => i * j);
}

bool isPalindrome(int n) {
  var s = n.toString();
  return s == s.reversed;
}

void main() {
  print(productsOfThreeDigitNumbers().where(isPalindrome).max());
}
